packages/cli/src/lib/flattenMetadata.ts

```diff
@@ -0,0 +1,34 @@
+import { Metadata } from '@appland/models';
+
+/** Flattens metadata into a string-string map suitable for use in telemetry.
+ * Ignores git, exception and fingerprints.
+ */
+export default function flattenMetadata(metadata: Metadata): Map<string, string> {
+  const result = new Map<string, string>();
+
+  if (metadata.app) result.set('app', metadata.app);
+
+  if (metadata.client) {
+    result.set('client.name', metadata.client.name);
+    result.set('client.url', metadata.client.url);
+    if (metadata.client.version) result.set('client.version', metadata.client.version);
+  }
+
+  if (metadata.frameworks)
+    for (const { name, version } of metadata.frameworks) result.set(`framework.${name}`, version);
+
+  if (metadata.labels) result.set('labels', metadata.labels.join(', '));
+
+  if (metadata.language) {
+    result.set('language.name', metadata.language.name);
+    result.set('language.version', metadata.language.version);
+    if (metadata.language.engine) result.set('language.engine', metadata.language.engine);
+  }
+
+  if (metadata.name) result.set('name', metadata.name);
+  if (metadata.recorder?.name) result.set('recorder.name', metadata.recorder.name);
```

Comment:
Please add recorder.type to this - new in 1.9.0

Reply:
Oh, ok. It seems the type in models needs updating in this case. We really should try to keep them up to date.